app/src/main/res/layout/component_recently_closed.xml

```diff
@@ -18,7 +18,8 @@
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/recently_closed_list"
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toBottomOf="parent"
```

Comment:
I was just wondering if was too much to ask to fix this issue since this PR is already in this area. 😄

Reply:
Of course! Are you referring to the weird animation when deleting items in the **bookmark** screen?

Reply:
I think this change solves the problem I was seeing with the bottom RecyclerView cutting off the overscroll animation. 🙂 

Thanks!

Reply:
I hadn't even noticed the over scroll problem. I'm glad this solves it. 

However, I was referring to a different problem. It's very subtle, but notice how the last item disappears and reappears when the second-to-last item is removed. 

https://user-images.githubusercontent.com/14791787/119891857-62fd4400-bf31-11eb-8a38-0e26a35e7868.mp4

This is the expected behavior:

https://user-images.githubusercontent.com/14791787/119891754-44974880-bf31-11eb-9004-923d5fcdd968.mp4

Reply:
I think this is the cause of `notifyItemRangeChanged`. Using `submitList` should be enough with the `ListAdapter` to know and calculate which items to animate. Nice catch!

Reply:
This problem was actually caused by the fact that the Recyclerview height was set to wrap_content. The bookmark screen also has this problem.

Reply:
Fair point! Let's remove the extra notify though, it's definitely doing double work there.

From the [`submitList`](https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter#submitList(java.util.List%3CT%3E)) docs you can see that they do the notifying for you:

> Submits a new list to be diffed, and displayed.
> If a list is already being displayed, a diff will be computed on a background thread, which will dispatch Adapter.notifyItem events on the main thread.

Reply:
Diff only checks the submitted list of `RecoverableTab`s. It does not, however, check the list of selected tabs for changes. Therefore, notifyItemRangeChanged must be called to update the RecyclerView whenever the selection has changed. 

For this to work with diff, the selection must be integrated into the `RecoverableTab` model, for example by adding an `isSelected` flag.

But, as you mentioned, it's still doing double work.

Reply:
Hmm yeah, that's a good point. Thank you for clarifying that for me. :)